### PR TITLE
feat: add prettier config support

### PR DESCRIPTION
**Pull request by kris-ellery**
- Added `prettierConfig`, which extends `prettier-config-zillow`
- Added `prettier.config.js` to base template
- Updated readme and migration guide

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,6 +122,19 @@ module.exports = {
 };
 ```
 
+## Modify Prettier Config
+
+You can customize Prettier configuration to your requirements. In example, you can update `prettier.config.js` with new `semi` value.
```
Comment (stonebk) [inline README.md:127]: I think this conflicts with `eslint-config-zillow` 🤔

Comment (kris-ellery) [reply README.md:127]: Ahh, it does. Auto-fix on save works, but then I get error from ESLint. I can try another setting or I can mention to use in tandem with ESLint config.

Comment (stonebk) [reply README.md:127]: Yeah, I think in general eslint config is used and then the prettier file is just a redirect:
https://github.com/zillow/javascript/tree/latest/packages/eslint-plugin-zillow#prettier-editor-plugin-integration

Comment (kris-ellery) [reply README.md:127]: Got it. I'll remove this section and info about modifying rules.

File: templates/base/CONTRIBUTING.md [hunk 2]
```diff
@@ -4,24 +4,24 @@
...
-- `npm run test:watch` will run the tests on every change.
+-   `npm run test:watch` will run the tests on every change.
 
 ## Building
 
-- `npm run build` will build the component for publishing to npm and also bundle the demo app.
+-   `npm run build` will build the component for publishing to npm and also bundle the demo app.
 
-- `npm run clean` will delete built resources.
+-   `npm run clean` will delete built resources.
```
Comment (stonebk) [inline templates/base/CONTRIBUTING.md:27]: What happened with all these spaces?

**Review by stonebk — APPROVED** [review]
Looks good!

**Pull request merged**